src/ansi-c/parser_static.inc

```diff
@@ -115,13 +115,13 @@ static void merge_types(irept &dest, irept &src)
   if(dest.id()!=ID_merged_type)
   {
     source_locationt location=static_cast<const exprt &>(dest).source_location();
-    typet new_type(ID_merged_type);
+    type_with_subtypest new_type(ID_merged_type);
```

Comment:
I'd suggest that this be merged after #3018 and thus `merged_typet` be used here.